internal/db/schema/migrations/oss/postgres/70/01_targets.up.sql

```diff
@@ -0,0 +1,40 @@
+begin;
+
+  create table target_address (
+    public_id wt_public_id,
+    address text not null,
```

Comment:
Are there any additional checks we can do on the address column?

Reply:
I added the same constraints that are applied to a host address. Are there any other constraints that should be added?

```
constraint address_must_be_more_than_2_characters
check(length(trim(address)) > 2)
constraint address_must_be_less_than_256_characters
check(length(trim(address)) < 256)
```

Reply:
I should also make public_id an immutable column.

Reply:
Also, if a target_address is removed, then the session should be canceled.

Reply:
Another potential solution is to create a new table called session_target_address, which can have a foreign key constraint to target_address. Then we can follow the same path of cancelling the session when the referenced column is set to null during a cascading delete.

Reply:
I'm starting to like the second solution more because we need to do the same for host_set since this relationship to a session is no longer required. This requires a few updates to other resources like the session_list view and the column `host_key` in the table `wh_session_connection_accumulating_fact`.